# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1444475271 -7200
#      Sat Oct 10 13:07:51 2015 +0200
# Node ID 69270256b879e3e85a7acc9f90e273a93e5cf925
# Parent  99eb26310cb335073e638dc0d95fbb4b23e00b27
Qt Figure: allow displaying coordinates when hovered axes parent is a uipanel.

* Canvas/cc (Canvas::canvasMouseMoveEvent): call "select_object" with the actual Canvas object which may also be a uipanel.

diff --git a/libgui/graphics/Canvas.cc b/libgui/graphics/Canvas.cc
--- a/libgui/graphics/Canvas.cc
+++ b/libgui/graphics/Canvas.cc
@@ -521,23 +521,23 @@ Canvas::canvasMouseMoveEvent (QMouseEven
               updateCurrentPoint (figObj, obj, event);
               gh_manager::post_callback (figObj.get_handle (),
                                          "windowbuttonmotionfcn");
             }
         }
     }
 
   // Update mouse coordinates in the figure window status bar
-  graphics_object figObj =
-    gh_manager::get_object (m_handle).get_ancestor ("figure");
+  graphics_object obj = gh_manager::get_object (m_handle);
+  graphics_object figObj = obj.get_ancestor ("figure");
 
-  if (figObj.valid_object ())
+  if (figObj.valid_object () && obj.valid_object ())
     {
       graphics_object currentObj, axesObj;
-      select_object (figObj, event, currentObj, axesObj, true);
+      select_object (obj, event, currentObj, axesObj, true);
 
       if (axesObj.valid_object ())
         {
           // FIXME: should we use signal/slot mechanism instead of
           //        directly calling parent fig methods
           Figure* fig =
             dynamic_cast<Figure*> (Backend::toolkitObject (figObj));
           axes::properties& ap = Utils::properties<axes> (axesObj);
